### PR TITLE
snappy: move gc to remove all but the latest snap

### PR DESCRIPTION
**Pull request by chipaca**
I expect the integration tests to fail, but as they won't run locally this is the only way I know to have them run. Please ignore until they pass...

### CODE REVIEW

**Comment by niemeyer** [comment]
LGTM, assuming they pass.

**Comment by mvo5** [comment]
retest this please

**Comment by zyga** [comment]
This fails on: 
```
Generated 'hello-world_6.0+fake1_all.snap' snap
sudo rm -rf /var/tmp/snap-build-809896684
sudo TMPDIR=/var/tmp SNAPPY_FORCE_CPI_URL=http://localhost:11028 snappy -test.run=^TestRunMain$ -test.coverprofile=/tmp/adt-run.3I6wmb/command1-artifacts/coverage/910807294 update

sudo rm -rf /var/tmp/snap-fake-store-blobs-368940929
/home/jenkins-slave/workspace/github-snappy-integration-tests-cloud/src/github.com/ubuntu-core/snappy/integration-tests/tests/update_app_test.go:48:
    ...open /home/jenkins-slave/workspace/github-snappy-integration-tests-cloud/src/github.com/ubuntu-core/snappy/integration-tests/tests/update_app_test.go: no such file or directory
... value string = "\n"
... regex string = "(?ms).*^hello-world.*fake1.*"

sudo snappy -test.run=^TestRunMain$ -test.coverprofile=/tmp/adt-run.3I6wmb/command1-artifacts/coverage/407538245 remove hello-world.canonical
Removing hello-world.canonical
START: <autogenerated>:125: updateAppSuite.TearDownTest
PASS: <autogenerated>:125: updateAppSuite.TearDownTest	0.000s

FAIL: /home/jenkins-slave/workspace/github-snappy-integration-tests-cloud/src/github.com/ubuntu-core/snappy/integration-tests/tests/update_app_test.go:39: updateAppSuite.TestAppUpdate
```

and on:

```
sudo rm -rf /var/tmp/snap-fake-store-blobs-000888594
/home/jenkins-slave/workspace/github-snappy-integration-tests-cloud/src/github.com/ubuntu-core/snappy/integration-tests/tests/update_os_test.go:83:
    ...open /home/jenkins-slave/workspace/github-snappy-integration-tests-cloud/src/github.com/ubuntu-core/snappy/integration-tests/tests/update_os_test.go: no such file or directory
... value string = "\n"
... regex string = "(?ms).*^Reboot to use ubuntu-core version .*\\.\n"

START: <autogenerated>:129: updateOSSuite.TearDownTest
PASS: <autogenerated>:129: updateOSSuite.TearDownTest	0.000s

FAIL: /home/jenkins-slave/workspace/github-snappy-integration-tests-cloud/src/github.com/ubuntu-core/snappy/integration-tests/tests/update_os_test.go:77: updateOSSuite.TestUpdateToSameReleaseAndChannel
```

**Comment by elopio** [comment]
@chipaca the problem is snap_remote_repo.go/SnapUpdates. It's failing to decode the data returned by the store, so the updateData var is empty and then snappy reports that it didn't find any updates.

What I fail to see is how can this happen. I don't see in your changes anything that could possibly affect the response from the fake store, nor the decoding of it.

File: snappy/rollback.go [hunk 1]
```diff
@@ -31,25 +30,5 @@ import (
 //
 // The version needs to be installed on disk
 func Rollback(pkg, ver string, inter progress.Meter) (version string, err error) {
```
Comment (mvo5) [inline snappy/rollback.go]: This really concerns me. I know this is mean to be temporary, but it is still making me really nervous. The rollback story is the biggest feature compared to traditional systems. Combined with the boot logic to auto rollback for broken boots its a very strong story. Breaking that even temporary does not feel right. Sorry for this negative review.

File: snappy/install.go [hunk 2]
```diff
@@ -268,10 +257,15 @@ func GarbageCollect(name string, flags InstallFlags, pb progress.Meter) error {
 		return nil
 	}
 
-	for _, snap := range snaps[:active-1] {
+	for i, snap := range snaps {
```
Comment (mvo5) [inline snappy/install.go]: If I'm reading this correct we remove all except the currently active snap. And this code runs after the new snap got installed. So there is only a single version installed afterwards(?). If that is the case we break rollback of broken OS/kernel updates here ? or am I missing something and I misremember the order?

Comment (chipaca) [inline snappy/install.go]: that's exactly what it's doing

**Pull request closed without merging**